routers/private/hook.go

```diff
@@ -434,6 +434,18 @@ func HookPostReceive(ctx *macaron.Context, opts private.HookOptions) {
 		}
 	}
 
+	// Push Options
+	if repo != nil {
+		repo.IsPrivate = opts.GitPushOptions.Bool(private.GitPushOptionRepoPrivate, repo.IsPrivate)
```

Comment:
Don't update the repository always, only update them when the env exist in fact. But I think we haven't check that.

Reply:
I'm now checking if the options map is `>0`

This isn't perfect because _any_ pushed option would cause a repo update, however the alternative is to explicitly match the incoming push options with a list of existing keys.  
Currently if someone pushes an incorrect push option, worst case is the repo would be updated to its same settings and nothing changes.